backend/functions/lib/adapters/gateways/dxtr/offices.dxtr.gateway.ts

```diff
@@ -103,7 +97,13 @@ export default class OfficesDxtrGateway implements OfficesGatewayInterface {
     );
 
     if (queryResult.success) {
-      return (queryResult.results as mssql.IResult<OfficeDetails>).recordset[0];
+      const results = queryResult.results as mssql.IResult<OfficeDetails>;
+      if (results.recordset.length === 0) {
+        throw new CamsError(MODULE_NAME, {
+          message: 'Office not found by query designator.',
+        });
+      }
+      return results.recordset[0];
```

Comment:
What about the case where we have multiple offices with the groupDesignator?

Reply:
The relationship between office and group designator is one-to-many. So each group designator will return only one office.